### DISCUSSION TITLE
brew upgrade --interactive?

### DISCUSSION THREAD

**Discussion by blogabe** [Homebrew/discussions#3786] [Everyday usage]
I understand 'brew upgrade --interactive' has a very specific meaning that targets only formulae. What I'd like to know is if there is a similar brew option to interactively choose which outdated formulae or casks get upgraded?

Presumably I can pipe 'brew outdated' into an interactive shell script to enable this type of functionality, but hoping there's a way natively within brew.

I searched for a bit before opening this up so my apologies if I didn't find a current conversation on this.

Thank you

**Comment by carlocab** [comment]
There is no way to do this natively with `brew`. But if you have `fzf` installed, you can do something like
```sh
outdated=$(brew outdated); [[ -n "$outdated" ]] && fzf --multi <<< $outdated | xargs brew upgrade
```
to do what you're looking for.

**Comment by blogabe** [comment]
Thank you @carlocab This seems like the easiest path forward.

I'm wondering if the HB maintainers are open to development on this front or might there be a reason why this has not yet been implemented? I'm not suggesting the maintainers develop it, simply if they'd be open to it.

**Reply by gromgit** [reply]
> I'm wondering if the HB maintainers are open to development on this front or might there be a reason why this has not yet been implemented? I'm not suggesting the maintainers develop it, simply if they'd be open to it.

The general answer to this seems to be "feature requests are better presented as actual PRs, because the devil really is in the unstated details of the request".

As for why such a feature doesn't yet exist...I'm not a maintainer, but in my experience, upgrade scenarios generally fall into one of the following categories:
1. a handful of formulae, which can easily be specified manually: `brew upgrade <formula> ...`
2. a medium-sized list, "but I just need to compile a Go program, so I don't care about the rest for now"...see [1].
3. a HUGE list (because no upgrades for a long time), in which case presenting them in a list for selection still makes users just give up on the upgrade entirely, or consider what they need "right now"...see [2]. Also, because of https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff, selecting just _one_ formula in this scenario will probably still upgrade "literally everything I've ever installed WTF!!!" (because no upgrades for a long time).

**Reply by carlocab** [reply]
`brew` doesn't really expose an interactive UI anywhere (aside from the occasional `sudo` prompt, but even that comes from a third-party installer that `brew` runs when installing some casks), so my guess is that this isn't really in scope for `brew`.

You could probably write an [external command](https://docs.brew.sh/Manpage#custom-external-commands) that does this, though. See https://github.com/buo/homebrew-cask-upgrade for an external tap that does what you're looking for for casks.

**Comment by blogabe** [comment]
Oh I've been using buo's tap for a few years now :)

As the upgrade of casks has improved over the years, I'm looking to do this natively in brew. While I didn't think it was an existing feature, I had thought to pipe the command to a script, but your earlier suggestion makes things easier. I'll either continue using buo's tap or use fzf.
